docker-images/al2/Dockerfile

```diff
@@ -0,0 +1,89 @@
+ARG AL_VERSION=2
+
+FROM amazonlinux:${AL_VERSION}
+
+WORKDIR /opt/s2n
+
+ARG OPENSSL_VERSION=openssl-1.1.1
+ARG GCC_VERSION=7
+ARG ZSH_THEME=cypher
+
+ENV S2N_LIBCRYPTO=${OPENSSL_VERSION} \
+    GCC_VERSION=${GCC_VERSION} \
+    BUILD_S2N=true \
+    TESTS=integration \
+    EPEL_REPO="https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm"
+
+# The `s2n_setup_env` assumes bash, not sh
+SHELL ["/bin/bash", "-c"]
+
+# set up user account
+RUN set -ex \
+    && amazon-linux-extras install ruby2.6 rust1 \
+    && yum install -y openssh-clients \
+    && mkdir ~/.ssh \
+    && touch ~/.ssh/known_hosts \
+    && ssh-keyscan -t rsa,dsa -H github.com >> ~/.ssh/known_hosts \
+    && chmod 600 ~/.ssh/known_hosts \
+    && yum install -y $EPEL_REPO \
+    && rpm --import https://download.mono-project.com/repo/xamarin.gpg \
+    && curl https://download.mono-project.com/repo/centos7-stable.repo | tee /etc/yum.repos.d/mono-centos7-stable.repo \
+    && amazon-linux-extras enable corretto8 \
+    && yum groupinstall -y "Development tools" \
+    && yum install -y GeoIP-devel ImageMagick asciidoc bzip2-devel bzr bzrtools cvs cvsps \
+    docbook-dtds docbook-style-xsl dpkg-dev e2fsprogs expat-devel expect fakeroot \
+    glib2-devel groff gzip icu iptables jq krb5-server libargon2-devel \
+    libcurl-devel libdb-devel libedit-devel libevent-devel libffi-devel \
+    libicu-devel libjpeg-devel libpng-devel libserf libsqlite3x-devel \
+    libtidy-devel libunwind libwebp-devel libxml2-devel libxslt libxslt-devel \
+    libyaml-devel libzip-devel mariadb-devel mercurial mlocate mono-devel \
+    ncurses-devel oniguruma-devel openssl openssl-devel perl-DBD-SQLite \
+    perl-DBI perl-HTTP-Date perl-IO-Pty-Easy perl-TimeDate perl-YAML-LibYAML \
+    postgresql-devel procps-ng python-configobj readline-devel rsync shadow-utils sgml-common \
+    subversion-perl sudo tar tcl tk vim wget which xfsprogs xmlto xz-devel zsh \
+    && gem install bundler;
+RUN useradd -m s2n-dev
+RUN  set -ex \
+  echo "s2n-dev ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/s2n-dev; \
```

Comment:
We should probably move this into a script that gets added so it stays consistent between images

Reply:
addd to bin and updated both al2 and ubuntu